src/video/sdl_v.cpp

```diff
@@ -774,10 +774,20 @@ bool VideoDriver_SDL::ToggleFullscreen(bool fullscreen)
 	if (_draw_mutex != NULL) _draw_mutex->BeginCritical(true);
 
 	/* Remember current window size */
-	if (fullscreen) SDL_GetWindowSize(_sdl_window, &_window_size_w, &_window_size_h);
+	if (fullscreen) {
+		SDL_GetWindowSize(_sdl_window, &_window_size_w, &_window_size_h);
+
+		/* Find fullscreen window size */
+		SDL_DisplayMode dm;
+		if (SDL_GetCurrentDisplayMode(0, &dm) < 0) {
```

Comment:
@PeterN this section makes ToggleFullscreen work with `SDL_WINDOW_FULLSCREEN`. Previously, I was a little confused on the expected behavior here: when in windowed mode, ToggleFullscreen should actually increase the window size to be the same as the current resolution, rather than just forcing a fullscreen on the current window size.